django/db/models/indexes.py

```diff
@@ -0,0 +1,57 @@
+
+__all__ = ['Index']
+
+
+class Index(object):
+
+    def __init__(self, *fields, **kwargs):
+        self.fields = fields
+        if 'model' in kwargs:
+            self.model = kwargs['model']
+        if 'name' in kwargs:
+            self.name = kwargs['name']
+
+    def as_sql(self, schema, suffix=''):
+        columns = [field for field in self.fields]
+        fields = [self.model._meta.get_field(field) for field in self.fields]
+        if len(fields) == 1 and fields[0].db_tablespace:
+            tablespace_sql = schema.connection.ops.tablespace_sql(fields[0].db_tablespace)
+        elif self.model._meta.db_tablespace:
+            tablespace_sql = schema.connection.ops.tablespace_sql(self.model._meta.db_tablespace)
+        else:
+            tablespace_sql = ""
+        if tablespace_sql:
+            tablespace_sql = " " + tablespace_sql
+
+        if not hasattr(self, 'name'):
+            self.name = schema._create_index_name(self.model, columns, suffix=suffix)
```

Comment:
I think the name should be generated as part of the index class and should be database backend independent, otherwise your migrations would yield different names depending on the database used to generate the migrations.